KSQL-1494: Test permutations of optimizations on and off

bbejeck opened the pull request:
### Description 
I changed the `QueryTranslationTest` to now run all tests with optimizations set to `all`  and `none`.  I also made a minor change to `SchemaTranslationTest` as it uses `EndToEndEngineTestUtil` as well.

We'll need to discuss how to approach the permutations as more optimization versions are available.  Each new version will require an additional test method (very low overhead to add), but increases the total test count by 128.  While that seems acceptable for now, more permutations across versions will quickly explode the test count making it somewhat of a burden to run as unit test each time.

### Testing done 
I ran the existing `QueryTranslationTest` which now runs 256 tests, 128 with optimizations off and 128 with optimizations on.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

bbejeck commented on ksql-engine/src/test/java/io/confluent/ksql/SchemaTranslationTest.java:
```diff
@@ -47,7 +47,7 @@ public SchemaTranslationTest(final String name, final Query query) {
   @Test
   public void shouldBuildAndExecuteQueries() {
-    EndToEndEngineTestUtil.shouldBuildAndExecuteQuery(this.query);
+    EndToEndEngineTestUtil.shouldBuildAndExecuteQuery(this.query, "none");
```
I simply added a hard-coded value here with optimizations turned off.  My thinking is just adding the parameter was good enough instead refactoring out a separate method at this point.

guozhangwang replied:
If we test with optimization turned on, do we need to manually change this value in the test code and re-run?

guozhangwang commented:
ping @confluentinc/ksql  team to review as well.

bbejeck commented:
Thinking about this some more, it makes sense to parameterize `SchemaTranslationTest` in the same way as `QueryTranslationTest`.

Also rebased with master.

\cc @guozhangwang

bbejeck commented:
Cleaned up the file generation by closing the `KsqlEngine` after writing each topology and rebased from master with new updates.

ping @big-andy-coates @rodesai for reviews

bbejeck commented:
updated again to use try-with-resources on `KsqlEngine` for file generation

bbejeck commented:
ping @big-andy-coates @rodesai for reviews this one should be quick; it has very small changes

bbejeck commented:
rebased this
